File: Java_Backend/E-Tour/src/main/java/entity/Customer.java
package entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer custId;
    
    private String custName;
    private String custEmail;
    private String custPhone;
    private LocalDate custDob;
    private Character custGender;
    private String custAddress;
    private String custCity;
    private String custState;
    private String custPincode;
    private String custCountry = "India";
    private LocalDateTime custCreatedAt = LocalDateTime.now();
    
    @OneToOne(mappedBy = "customer", cascade = CascadeType.ALL)
    private Authentication authentication;
    

    
    @OneToMany(mappedBy = "customer", cascade = CascadeType.ALL)
    private List<Booking> bookings;
}

File: DotNet_Backend/SafarnamaApplication/SafarnamaApplication/Models/Customer.cs
using System;
using System.Collections.Generic;

namespace SafarnamaApplication.Models;

public partial class Customer
{
    public int CustId { get; set; }

    public string? CustName { get; set; }

    public string? CustEmail { get; set; }

    public string? CustPhone { get; set; }

    public DateOnly? CustDob { get; set; }

    public string? CustGender { get; set; }

    public string? CustAddress { get; set; }

    public string? CustCity { get; set; }

    public string? CustState { get; set; }

    public string? CustPincode { get; set; }

    public string? CustCountry { get; set; }

    public DateTime? CustCreatedAt { get; set; }

    public virtual ICollection<Authentication> Authentications { get; set; } = new List<Authentication>();

    public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();
}
